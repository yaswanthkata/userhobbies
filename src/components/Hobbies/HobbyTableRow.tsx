import React from "react";
import { Hobby } from "../../types";

interface IHobbyTableRowProps {
  hobby: Hobby;
  onRemove(hobby: Hobby): void;
}

const HobbyTableRow = ({ hobby, onRemove }: IHobbyTableRowProps) => {
  const getPassion = (level: number) => {
    switch (level) {
      case 1:
        return "Low";
      case 2:
        return "Medium";
      case 3:
        return "High";
      case 4:
        return "Very High";
      default:
        return "";
    }
  };
  return (
    <tr>
      <td>{hobby.passion && "Passion : " + getPassion(hobby.passion)}</td>
      <td>
        <h2>{hobby.name}</h2>
      </td>
      <td>{"Since " + hobby.year}</td>
      <td>
        <button
          className="btn btn-red"
          onClick={() =>
            window.confirm("Are you sure you wish to delete the hobby?") &&
            onRemove(hobby)
          }
        >
          {" "}
          <i className="fa fa-trash" aria-hidden="true" />
        </button>
      </td>
    </tr>
  );
};

export default HobbyTableRow;
