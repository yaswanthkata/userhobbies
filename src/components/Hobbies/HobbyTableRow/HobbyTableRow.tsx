import React from "react";
import { Hobby } from "../../../types";

interface IHobbyTableRowProps {
  hobby: Hobby;
  onRemove(hobby: Hobby): void;
}

const HobbyTableRow = ({ hobby, onRemove }: IHobbyTableRowProps) => {
  const getPassionDescription = (level: number) => {
    switch (level) {
      case 1:
        return "Low";
      case 2:
        return "Medium";
      case 3:
        return "High";
      case 4:
        return "Very High";
      default:
        return "";
    }
  };
  return (
    <tr>
      <td>
        {hobby.passion && "Passion : " + getPassionDescription(hobby.passion)}
      </td>
      <td>
        <h2>{hobby.name}</h2>
      </td>
      <td>{hobby.year && "Since " + hobby.year}</td>
      <td>
        <button
          className="btn btn-red"
          onClick={() =>
            window.confirm("Are you sure you wish to delete the hobby?") &&
            onRemove(hobby)
          }
        >
          {" "}
          <i className="fa fa-trash" aria-hidden="true" />
        </button>
      </td>
    </tr>
  );
};

export default HobbyTableRow;
