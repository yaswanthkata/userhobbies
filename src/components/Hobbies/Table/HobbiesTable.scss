
@import '../../../styles/variables';
table {
  border: solid 1px $color-border-light;
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
  overflow: auto;

  tbody td {
    border-bottom: solid 1px $gray;
    padding: 10px;
  }
}