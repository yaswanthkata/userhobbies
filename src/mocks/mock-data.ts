import { Hobby } from "./../types";
import { User } from "./../types";

export const hobbies: Hobby[] = [
  { id: 1, name: "Swimming", passion: 0, year: 2008, userId: 1 },
  { id: 2, name: "Cycling", passion: 1, year: 2009, userId: 1 }
];
export const users: User[] = [
  { id: 1, name: "Yaswanth kata", thumbnail: "" },
  { id: 2, name: "Pulkit", thumbnail: "" }
];
