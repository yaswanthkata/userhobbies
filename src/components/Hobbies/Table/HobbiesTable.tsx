import React from "react";
import { Hobby } from "../../../types";
import HobbyTableRow from "../HobbyTableRow/HobbyTableRow";
import "./HobbiesTable.scss";

export interface IHobbiesTableProps {
  hobbies: Hobby[];
  onRemove(id: number): void;
}

const HobbiesTable = ({ hobbies, onRemove }: IHobbiesTableProps) => {
 
  const removeHobby = (hobby: Hobby) => {
    onRemove(hobby.id);
  };

  return (
    <table>
      <colgroup>
        <col width="25%" />
        <col width="45%" />
        <col width="25%" />
        <col width="5%" />
      </colgroup>
      <tbody>
        {hobbies.map(hobby => (
          <HobbyTableRow key={hobby.id} hobby={hobby} onRemove={removeHobby} />
        ))}
      </tbody>
    </table>
  );
};

export default HobbiesTable;
