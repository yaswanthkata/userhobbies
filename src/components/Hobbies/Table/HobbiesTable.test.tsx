import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import HobbiesTable, { IHobbiesTableProps } from './HobbiesTable';
import * as mockData from "../../../mocks/mock-data";
import HobbyTableRow from '../HobbyTableRow/HobbyTableRow';

const props: IHobbiesTableProps = {
    hobbies: mockData.hobbies,
    onRemove: () => jest.fn()
}

let componentWrapper: ShallowWrapper<React.Component>;

beforeEach(() => {
    componentWrapper = shallow(<HobbiesTable {...props} />);
});

it('should render hobbies table component correctly', () => {
    expect(componentWrapper).toMatchSnapshot();
});

it('should render one <table> element', () => {
    expect(componentWrapper.find('table')).toHaveLength(1);
});

it("should render  a list of rows with hobbies", () => {
    const numberOfHobbyRows = componentWrapper.find(HobbyTableRow).length;

    expect(numberOfHobbyRows).toEqual(2);
});