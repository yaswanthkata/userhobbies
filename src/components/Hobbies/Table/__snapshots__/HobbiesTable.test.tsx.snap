// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render hobbies table component correctly 1`] = `ShallowWrapper {}`;
